version: '3.7'
services: 
  db:
    image: postgres:latest
    ports:
      - '5432:5432'
    environment:
      POSTGRES_USER: postgres
      POSTGRES_DB: filebox
      POSTGRES_PASSWORD: password
      TZ: 'GMT+8'
      PGTZ: 'GMT+8'
    networks:
      - filebox-net

  redis:
    image: redis:latest
    restart: always
    ports:
      - '6379:6379'
    networks:
      - filebox-net

  filebox-server:
    build: ../server/dockerfile
    depends_on:
      db:
        condition: service_started
      redis:
        condition: service_started
    networks:
      - filebox-net
    ports:
      - "8888:8888"
    expose:
      - 8888
    environment:
      DATABASE_URL: 'postgres://postgres:password@db/filebox'
      HTTP_SERVER_ADDR: '0.0.0.0:8888'
      UPLOAD_FILE_PATH: 'uploaded'
      GRACEFUL_SHUTDOWN_TIMEOUT_SEC: 5
      REDIS_CONN_ADDR: 'redis:6379' # actix-redis does not accept Redis URL starting with redis:// for some reason. https://stackoverflow.com/questions/73780056/dns-resolver-failed-to-resolve-host-redis-failed-to-lookup-address-informati
      CODE_LEN: 5
      IP_VISIT_ERROR_LIMIT: 5
      IP_UPLOAD_LIMIT: 5
      IP_VISIT_ERROR_DURATION_DAY: 1
      ALLOWED_ORIGIN: 'http://127.0.0.1:5173'
  
  filebox-web:
    build: ../web/dockerfile
    depends_on:
      filebox-server:
        condition: service_started
    networks:
      - filebox-net
    ports:
      - "80:80"
    expose:
      - 80

networks:
  filebox-net:
